packages/astro/src/core/dev/dev.ts

```diff
@@ -32,7 +34,26 @@ export interface DevServer {
 export default async function dev(
 	settings: AstroSettings,
 	options: DevOptions
-): Promise<DevServer> {
+): Promise<DevServer | undefined> {
+	if (options.flags?.help || options.flags?.h) {
```

Comment:
don't you need to duplicate this logic elsewhere?

most of the if statements don't check for `-h`

Reply:
I see what you mean. I had applied the suggestions, but then they were removed because I force pushed. Thank you for catching that, I'll fix the code ASAP!